plugins/processors/awsentity/entityattributes/entityattributes.go

```diff
@@ -0,0 +1,117 @@
+	AttributeEntityAutoScalingGroup      = AWSEntityPrefix + "auto.scaling.group"
+
+	// The following are possible platform values
+	AttributeEntityEC2Platform = "AWS::EC2"
+	AttributeEntityEKSPlatform = "AWS::EKS"
+	AttributeEntityK8sPlatform = "K8s"
+
+	// The following Fields are the actual names attached to the Entity requests.
+	ServiceName           = "Name"
+	DeploymentEnvironment = "Environment"
+	EntityType            = "Type"
+	ResourceType          = "ResourceType"
+	Identifier            = "Identifier"
+	AwsAccountId          = "AwsAccountId"
+	EksCluster            = "EKS.Cluster"
+	K8sCluster            = "K8s.Cluster"
+	NamespaceField        = "K8s.Namespace"
+	Workload              = "K8s.Workload"
+	Node                  = "K8s.Node"
+	ServiceNameSource     = "AWS.ServiceNameSource"
+	Platform              = "PlatformType"
+	InstanceID            = "EC2.InstanceId"
+	AutoscalingGroup      = "EC2.AutoScalingGroup"
+
+	// The following are values used for the environment fallbacks required on EC2
+	DeploymentEnvironmentFallbackPrefix = "ec2:"
+	DeploymentEnvironmentDefault        = DeploymentEnvironmentFallbackPrefix + "default"
+)
+
+// KeyAttributeEntityToShortNameMap is used to map key attributes from otel to the actual values used in the Entity object
+var keyAttributeEntityToShortNameMap = map[string]string{
+	AttributeEntityType:                  EntityType,
+	AttributeEntityResourceType:          ResourceType,
+	AttributeEntityIdentifier:            Identifier,
+	AttributeEntityAwsAccountId:          AwsAccountId,
+	AttributeEntityServiceName:           ServiceName,
+	AttributeEntityDeploymentEnvironment: DeploymentEnvironment,
+}
+
+// attributeEntityToShortNameMap is used to map attributes from otel to the actual values used in the Entity object
+var attributeEntityToShortNameMap = map[string]string{
+	AttributeEntityNamespace:         NamespaceField,
+	AttributeEntityWorkload:          Workload,
+	AttributeEntityNode:              Node,
+	AttributeEntityPlatformType:      Platform,
+	AttributeEntityInstanceID:        InstanceID,
+	AttributeEntityAutoScalingGroup:  AutoscalingGroup,
+	AttributeEntityServiceNameSource: ServiceNameSource,
+}
+
+func GetKeyAttributeEntityShortNameMap() map[string]string {
+	return keyAttributeEntityToShortNameMap
+}
+
+// Cluster attribute prefix could be either EKS or K8s. We set the field once at runtime.
+func GetAttributeEntityShortNameMap() map[string]string {
+	if _, ok := attributeEntityToShortNameMap[AttributeEntityCluster]; !ok {
+		attributeEntityToShortNameMap[AttributeEntityCluster] = clusterType()
+	}
+	return attributeEntityToShortNameMap
+}
+
+// Container Insights attributes used for scraping EKS related information
+const (
+	NodeName  = "NodeName"
+	Namespace = "Namespace"
+	// PodName in Container Insights is the workload(Deployment, Daemonset, etc) name
+	PodName = "PodName"
+)
+
+func clusterType() string {
+	ctx := context.CurrentContext()
+	mode := ctx.KubernetesMode()
```

Comment:
Don't use these in non-translators. These are not guaranteed to be populated during the agent runtime.

Reply:
Blocking comment